### PR TITLE
rename: hosted_zone_preprocessor.rs to route53_resource_id_preprocess…

### PR DESCRIPTION
**Pull request by Velfi**
## Motivation and Context

[aws-sdk-rust#554](https://github.com/awslabs/aws-sdk-rust/issues/554)

## Description

rename: hosted_zone_preprocessor.rs to route53_resource_id_preprocessor.rs
update: trim_hosted_zone to work for more kinds of resource ids
add: new Route53 protocol test for GetChange resource id trimming
update: InlineDependency.forRustFile to accept filenames with a ".rs" extension

## Testing

I added a new protocol test

## Checklist

- [x] I have updated `CHANGELOG.next.toml` if I made changes to the smithy-rs codegen or runtime crates
- [x] I have updated `CHANGELOG.next.toml` if I made changes to the AWS SDK, generated SDK code, or SDK runtime crates

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: aws/sdk-codegen/src/main/kotlin/software/amazon/smithy/rustsdk/customize/route53/Route53Decorator.kt [hunk 1]
```diff
@@ -50,29 +52,39 @@ class Route53Decorator : RustCodegenDecorator {
...
 
-    private val trimZone =
-        RuntimeType.forInlineDependency(InlineAwsDependency.forRustFile("hosted_zone_preprocessor"))
-            .member("trim_hosted_zone")
+    private val trimResourceId =
+        RuntimeType.forInlineDependency(
+            InlineAwsDependency.forRustFile(
+                "route53_resource_id_preprocessor",
+                visibility = Visibility.PRIVATE,
+                CargoDependency.Regex,
```
[the remark this answers is not in the document]
Comment (rcoh) [reply aws/sdk-codegen/src/main/kotlin/software/amazon/smithy/rustsdk/customize/route53/Route53Decorator.kt:76]: cargo udeps should hopefully catch this

File: aws/sdk/aws-models/route53-tests.smithy [hunk 2]
```diff
@@ -18,3 +18,17 @@ apply ListResourceRecordSets @httpRequestTests([
...
+        documentation: "This test validates that change id is correctly trimmed",
+        method: "GET",
+        protocol: "aws.protocols#restXml",
+        uri: "/2013-04-01/change/SOMECHANGEID",
+        bodyMediaType: "application/xml",
+        params: {
+            "Id": "/change/SOMECHANGEID"
+        }
+    }
+])
```
Comment (rcoh) [inline aws/sdk/aws-models/route53-tests.smithy:34]: lets add protocol tests for all the supported prefixes

File: aws/sdk-codegen/src/main/kotlin/software/amazon/smithy/rustsdk/customize/route53/Route53Decorator.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (Velfi) [reply aws/sdk-codegen/src/main/kotlin/software/amazon/smithy/rustsdk/customize/route53/Route53Decorator.kt:76]: You are correct, I wonder why it didn't catch that.

**Comment by Velfi** [comment]
I had originally added the `resource` prefix because boto had it but I can't see anywhere in the model where that would be used so I've removed it. I also added a test for `delegationset` so now all 3 supported prefixes are supported

**Review by rcoh — APPROVED** [review]

**Pull request merged**